sources/osgShader/utils.js

```diff
@@ -107,23 +111,279 @@ var callFunction = function(funcName, output, inputs) {
         }
 
         for (var i = 0, l = inputs.length; i < l; i++) {
-            // check if it's a variable and not a constant
-            if (inputs[i].getVariable) callString += inputs[i].getVariable();
-            else callString += inputs[i];
-
+            callString += inputs[i].getVariable ? inputs[i].getVariable() : inputs[i];
             if (i !== l - 1) callString += ', ';
         }
     }
 
     callString += ' );\n';
 
-    if (osgShader.debugShaderNode) return debug.join('\n') + '\n' + callString;
+    if (osgShader.debugShaderNode) {
+        return debug.join('\n') + '\n' + callString;
+    }
 
     return callString;
 };
 
+var extractMapVariable = function(option) {
+    var res = option.match(/([^\s]+):([^\s]+)/g);
```

Comment:
regexp really takes some time to decrypt/read when you read  code, 
I love it very much when there is a sample of what it's supposed to match.
```
// ie myNodeName:myResult
 var res = option.match(/([^\s]+):([^\s]+)/g);
```